collect_app/src/main/java/org/odk/collect/android/tasks/DiskSyncTask.java

```diff
@@ -301,7 +305,11 @@ public ContentValues buildContentValues(File formDefFile) throws IllegalArgument
             updateValues.put(FormsColumns.JR_VERSION, version);
         }
         if (submission != null) {
-            updateValues.put(FormsColumns.SUBMISSION_URI, submission);
+            if(checkValidSubmissionURI(submission)) {
+                updateValues.put(FormsColumns.SUBMISSION_URI, submission);
+            } else {
+                throw new IllegalArgumentException(formDefFile.getName() + " :: bad url");
```

Comment:
Note that this is not required for all exception text. It's just that this particular exception text gets shown in the UI!